worlds/stardew_valley/logic.py

```diff
@@ -1,500 +1,534 @@
+                                 self.has(Mineral.jade) & self.has(Mineral.amethyst) & self.has_relationship(NPC.emily, 4) &
+                                 self.has(ArtisanGood.cloth) & self.can_reach_region(Region.haley_house),
+            SpecialOrder.gifts_for_george: self.has_season(Season.spring) & self.has(Forageable.leek),
+            SpecialOrder.fragments_of_the_past: self.can_reach_region(Region.dig_site),
+            SpecialOrder.gus_famous_omelet: self.has(AnimalProduct.any_egg),
+            SpecialOrder.crop_order: self.can_farm_perfectly(),
+            SpecialOrder.community_cleanup: self.can_crab_pot(),
+            SpecialOrder.the_strong_stuff: self.can_keg(Vegetable.potato),
+            SpecialOrder.pierres_prime_produce: self.can_farm_perfectly(),
+            SpecialOrder.robins_project: self.can_chop_perfectly() & self.has(Material.hardwood),
+            SpecialOrder.robins_resource_rush: self.can_chop_perfectly() & self.has(Fertilizer.tree) & self.can_mine_perfectly(),
+            SpecialOrder.juicy_bugs_wanted_yum: self.has(Loot.bug_meat),
+            SpecialOrder.tropical_fish: self.has_island_transport() & self.has(Fish.stingray) & self.has(Fish.blue_discus) & self.has(Fish.lionfish),
+            SpecialOrder.a_curious_substance: self.can_mine_perfectly() & self.can_mine_to_floor(80),
+            SpecialOrder.prismatic_jelly: self.can_mine_perfectly() & self.can_mine_to_floor(40),
+            SpecialOrder.qis_crop: self.can_farm_perfectly() & self.can_reach_region(Region.greenhouse) &
+                         self.can_reach_region(Region.island_west) & self.has_total_skill_level(50) &
+                         self.has(Machine.seed_maker),
+            SpecialOrder.lets_play_a_game: self.has_junimo_kart_max_level(),
+            SpecialOrder.four_precious_stones: self.has_lived_months(MAX_MONTHS) & self.has("Prismatic Shard") &
+                                    self.can_mine_perfectly_in_the_skull_cavern(),
+            SpecialOrder.qis_hungry_challenge: self.can_mine_perfectly_in_the_skull_cavern() & self.has_max_buffs(),
+            SpecialOrder.qis_cuisine: self.can_cook() & self.can_spend_money(250000),
+            SpecialOrder.qis_kindness: self.can_give_loved_gifts_to_everyone(),
+            SpecialOrder.extended_family: self.can_fish_perfectly() & self.has(Fish.angler) & self.has(Fish.glacierfish) &
+                               self.has(Fish.crimsonfish) & self.has(Fish.mutant_carp) & self.has(Fish.legend),
+            SpecialOrder.danger_in_the_deep: self.can_mine_perfectly() & self.has_mine_elevator_to_floor(120),
+            SpecialOrder.skull_cavern_invasion: self.can_mine_perfectly_in_the_skull_cavern() & self.has_max_buffs(),
+            SpecialOrder.qis_prismatic_grange: self.has(Loot.bug_meat) & self.can_spend_money(80000), # All colors can be bought except purple
```

Comment:
I believe this needs some kind of access requirements for the Ranch, Saloon, and Blacksmith